### PR TITLE
piston-cli: add module

### PR DESCRIPTION
**Pull request by ethancedwards8**
### Description

I added a module for piston-cli, I've also tested it
### Checklist

- [x] Change is backwards compatible.

- [x] Code formatted with `./format`.

- [x] Code tested through `nix-shell --pure tests -A run.all`.

- [ ] Test cases updated/added. See [example](https://github.com/nix-community/home-manager/commit/f3fbb50b68df20da47f9b0def5607857fcc0d021#diff-b61a6d542f9036550ba9c401c80f00ef).

- [x] Commit messages are formatted like

    ```
    {component}: {description}

    {long description}
    ```

    See [CONTRIBUTING](https://github.com/nix-community/home-manager/blob/master/doc/contributing.adoc#sec-commit-style) for more information and [recent commit messages](https://github.com/nix-community/home-manager/commits/master) for examples.

- If this PR adds a new module

  - [x] Added myself as module maintainer. See [example](https://github.com/nix-community/home-manager/blob/068ff76a10e95820f886ac46957edcff4e44621d/modules/programs/lesspipe.nix#L6).

  - [x] Added myself and the module files to `.github/CODEOWNERS`.
  N/A, I'm already a nixos maintainer.

### CODE REVIEW

**Review by MagicRB — APPROVED** [review]
At a glance, looks fine to me.

File: modules/programs/piston-cli.nix [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+      example = literalExample ''
+        {
+          # Code highlihting theme
+          theme = "emacs";
+          # Used for displaying your resultant output you get from the api
+          box_style = "MINIMAL_DOUBLE_HEAD";
+          # This is what is used for continuing your prompt when you go a new line
+          prompt_continuation = "...";
+          # This is the start symbol used when you are shell
+          prompt_start = ">>>";
```
Comment (berbiche) [inline modules/programs/piston-cli.nix:33]:
```suggestion
          theme = "emacs";
          box_style = "MINIMAL_DOUBLE_HEAD";
          prompt_continuation = "...";
          prompt_start = ">>>";
```

No need for the comments :smile:

File: modules/programs/piston-cli.nix [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+        }
+      '';
+      description = ''
+        Configuration written to
+        <filename>~/.config/piston-cli/config.yml</filename>.
```
Comment (berbiche) [inline modules/programs/piston-cli.nix:38]: Can you provide a link to the documentation as well?
If the documentation is only available in a manpage, you can reference it with the following syntax:
```
<citerefentry>
  <refentrytitle>command</refentrytitle>
  <manvolnum>1</manvolnum>
</citerefentry>
```
where command is the name of man page and 1 the type of pages.

**Review by berbiche — COMMENTED** [review]
Thanks for the contribution :smile: !

Would you mind removing the empty `flake.lock` file?

File: modules/programs/piston-cli.nix (hunk 2 above)
Comment (ethancedwards8) [reply modules/programs/piston-cli.nix:38]: Sadly the project has literally zero documentation. You have to sort through the code to find options. I listed all the possible (Yes, only four) in the example.

Comment (berbiche) [reply modules/programs/piston-cli.nix:34]: Feel free to add a link to the documentation once the upstream project gets one.

**Review by berbiche — APPROVED** [review]

**Comment by berbiche** [comment]
Can you rebase on master for the test to pass?

**Comment by ethancedwards8** [comment]
> Can you rebase on master for the test to pass?

I think I did that right
```
git clone https://github.com/nix-community/home-manager
cd home-manager/
git remote -v
git remote add ethan git@github.com:ethancedwards8/home-manager
git fetch ethan
git checkout module-programs-piston-cli 
git pull --rebase origin master
git push ethan module-programs-piston-cli -f
```

File: modules/programs/piston-cli.nix [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+      '';
+    };
+  };
+
+  config = mkIf cfg.enable {
+    home.packages = [ cfg.package ];
+
+    xdg.configFile."piston-cli/config.yml" = mkIf (cfg.settings != { }) {
+      source = yamlFormat.generate "config.yml" cfg.settings;
+    };
```
Comment (sumnerevans) [inline modules/programs/piston-cli.nix:44]: Does Piston support Darwin? If so, where does the config go?

File: modules/programs/piston-cli.nix [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+        {
+          theme = "emacs";
+          box_style = "MINIMAL_DOUBLE_HEAD";
+          prompt_continuation = "...";
+          prompt_start = ">>>";
+        }
+      '';
+      description = ''
+        Configuration written to
+        <filename>~/.config/piston-cli/config.yml</filename>.
```
Comment (sumnerevans) [inline modules/programs/piston-cli.nix:34]:
```suggestion
        <filename>$XDG_CONFIG_HOME/piston-cli/config.yml</filename>.
```

**Review by sumnerevans — CHANGES_REQUESTED** [review]
Looks good, overall!

File: modules/programs/piston-cli.nix (hunk 3 above)
Comment (ethancedwards8) [reply modules/programs/piston-cli.nix:44]: Yes, and it goes in the same place as linux.

**Pull request merged**